fuse_viz/src/serialized_graph_display.cpp

```diff
@@ -155,36 +197,105 @@ void SerializedGraphDisplay::processMessage(const fuse_msgs::SerializedGraph::Co
       continue;
     }
 
-    const auto& stamp = orientation->stamp();
-    const auto position_uuid = fuse_variables::Position2DStamped(stamp, orientation->deviceId()).uuid();
+    const auto position_uuid = fuse_variables::Position2DStamped(orientation->stamp(), orientation->deviceId()).uuid();
     if (!graph->variableExists(position_uuid))
     {
       continue;
     }
 
     const auto position = dynamic_cast<const fuse_variables::Position2DStamped*>(&graph->getVariable(position_uuid));
 
-    const tf2::Quaternion tf_q(tf2::Vector3(0, 0, 1), orientation->yaw());
+    variable_property_->createAndInsertOrUpdateVisual(scene_manager_, root_node_, *position, *orientation);
 
-    const Ogre::Vector3 p(position->x(), position->y(), 0);
-    const Ogre::Quaternion q(tf_q.w(), tf_q.x(), tf_q.y(), tf_q.z());
+    variables_changed_map_[position_uuid] = true;
+  }
 
-    // Add sphere:
-    Object* sphere = new rviz::Shape(rviz::Shape::Sphere, scene_manager_, variables_spheres_node_);
-    sphere->setPosition(p);
-    sphere->setScale(scale);
-    sphere->setColor(1.0, 0.0, 0.0, 1.0);
-    graph_shapes_.push_back(sphere);
+  for (const auto& constraint : graph->getConstraints())
+  {
+    const auto relative_pose = dynamic_cast<const fuse_constraints::RelativePose2DStampedConstraint*>(&constraint);
+    if (!relative_pose)
+    {
+      continue;
+    }
 
-    // Add axes:
-    Object* axes = new rviz::Axes(scene_manager_, variables_axes_node_, 10.0, 1.0);
-    axes->setPosition(p);
-    axes->setOrientation(q);
-    axes->setScale(scale);
-    graph_shapes_.push_back(axes);
+    const auto constraint_uuid = constraint.uuid();
+    const auto& constraint_source = constraint.source();
+
+    if (source_color_map_.find(constraint_source) == source_color_map_.end())
+    {
+      const auto n = source_color_map_.size() + 1;
```

Comment:
This produces a color map dynamically by computing the hue value for HSV color space based on the number of items. The sequence follow is: `1/2, 1/4, 3/4, 1/8, 3/8, 5/8, 7/8, ...`

I could encapsulate this in a class if you think it'd be cleaner.